compiler/src/main/java/com/squareup/anvil/compiler/ModuleMerger.kt

```diff
@@ -48,27 +45,29 @@ internal class ModuleMerger(
   override fun generateClassSyntheticParts(codegen: ImplementationBodyCodegen) {
     val module = moduleDescriptorFactory.create(codegen.descriptor.module)
     val clazz = codegen.descriptor.toClassReference(module)
-    val holder = AnnotationHolder.create(clazz) ?: return
-    val (mergeAnnotation, annotationClass, daggerClass, daggerFqName, modulesKeyword) = holder
 
-    if (clazz.isAnnotatedWith(daggerFqName)) {
-      throw AnvilCompilationExceptionClassReference(
-        message = "When using @${annotationClass.simpleName} it's not allowed to annotate the " +
-          "same class with @${daggerClass.simpleName}. The Dagger annotation will be generated.",
-        classReference = clazz
-      )
-    }
+    val annotations = clazz.annotations
+      .findAll(mergeComponentFqName, mergeSubcomponentFqName, mergeModulesFqName)
+      .ifEmpty { return }
 
-    val mergeScope = mergeAnnotation.scope()
-    val predefinedModules = mergeAnnotation.modules()
+    val daggerAnnotationClass = annotations[0].daggerAnnotationClass
+    val daggerModulesKeyword = annotations[0].modulesKeyword
+    val scopes = annotations.map { it.scope() }
+
+    val predefinedModules = annotations.flatMap {
+      it.argumentAt(it.modulesKeyword, index = 1)?.value<List<ClassReference>>().orEmpty()
+    }
     val anvilModuleName = createAnvilModuleName(clazz)
 
-    val modules = classScanner
-      .findContributedClasses(
-        module = module,
-        annotation = contributesToFqName,
-        scope = mergeScope.fqName
-      )
+    val contributesAnnotations = scopes
+      .flatMap { scope ->
```

Comment:
This was the general approach. Instead of querying a single scope, I had to flatmap the collections for multiple scopes.